### PR TITLE
zonemaster-cli: init at 4.0.1

### PR DESCRIPTION
**Pull request by qbit**
###### Description of changes

This PR adds zonemaster-cli and its dependencies.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -26049,6 +26091,44 @@ let
...
+    version = "4.5.1";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/Z/ZN/ZNMSTR/Zonemaster-Engine-v4.5.1.tar.gz";
+      sha256 = "45d204c6dad7cd90176084bf2427baa8ce503684a5699ebeb236e4d33bc0ba86";
+    };
+    buildInputs = [ PodCoverage TestDifferences TestException TestFatal TestNoWarnings TestPod ];
+    propagatedBuildInputs = [ ClassAccessor Clone EmailValid FileShareDir FileSlurp IOSocketInet6 ListMoreUtils ModuleFind Moose MooseXSingleton NetIP Readonly TextCSV ZonemasterLDNS libintlperl ];
+
+    # > t/Test-dnssec.t         (Wstat: 768 (exited 3) Tests: 763 Failed: 3)
+    doCheck = false;
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:26105]: Can this individual failing test be removed instead?

**Review by stigtsp — CHANGES_REQUESTED** [review]

**Review by stigtsp — COMMENTED** [review]
I think it would make more sense to add a `perlPackages.ZonemasterCLI` to `perl-packages.nix`, and then reference `zonemaster-cli = perlPackages.ZonemasterCLI` from `all-packages.nix`:
- It's follows the convention used by other cli tools that are served as CPAN modules like `ack`, `exiftool`, etc.
- Updates will be easier to handle since most CPAN modules are in `perl-packages.nix`

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (qbit) [reply pkgs/top-level/perl-packages.nix]: Updated to disable that test (and the manifest test as it checks for the Test-dnssec.t test :P)!

**Comment by qbit** [comment]
> I think it would make more sense to add a `perlPackages.ZonemasterCLI` to `perl-packages.nix`, and then reference `zonemaster-cli = perlPackages.ZonemasterCLI` from `all-packages.nix`:
> 
> * It's follows the convention used by other cli tools that are served as CPAN modules like `ack`, `exiftool`, etc.
> * Updates will be easier to handle since most CPAN modules are in `perl-packages.nix`

Updated to work this way. Thanks for pointing it out!

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -26049,6 +26091,90 @@ let
...
+      rm -f t/Test-dnssec.t
+
+      #   Failed test 'make distcheck gives empty output'
+      #   at t/manifest.t line 36.
+      #          got: 'No such file: t/Test-dnssec.t
+      # '
+      #     expected: ""
+      # Looks like you failed 1 test of 2.
+      rm -f t/manifest.t
+    '';
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:26151]: The large comment blocks with TAP failures are a bit too noisy imho, it's usually good enough to say that you're disabling network tests or something like that :-)

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -26049,6 +26091,90 @@ let
...
+      MooseXGetopt
+      TextReflow
+      ZonemasterEngine
+      ZonemasterLDNS
+      libintlperl
+    ];
+
+    preConfigure = ''
+      sed -i 's@/usr/bin/env perl@${perl}/bin/perl@g' script/*
+    '';
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:26112]: I believe patchShebangs should take care of this, if I'm not mistaken

File: pkgs/top-level/perl-packages.nix [hunk 4]
```diff
@@ -26049,6 +26091,90 @@ let
...
+      url = "mirror://cpan/authors/id/Z/ZN/ZNMSTR/Zonemaster-CLI-v4.0.1.tar.gz";
+      sha256 = "edd34f7b8137e492e6ce8474c45a550572dca5056abdefc45c076df9d6965ca0";
+    };
+    propagatedBuildInputs = [
+      JSONXS
+      MooseXGetopt
+      TextReflow
+      ZonemasterEngine
+      ZonemasterLDNS
+      libintlperl
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:26107]:
```suggestion
      libintl-perl
```

```
error: undefined variable 'libintlperl'

       at /home/sgo/.cache/nixpkgs-review/pr-187052/nixpkgs/pkgs/top-level/perl-packages.nix:26122:7:

        26121|       ZonemasterLDNS
        26122|       libintlperl
             |       ^
        26123|     ];
(use '--show-trace' to show detailed location information)

```

**Review by stigtsp — CHANGES_REQUESTED** [review]
Getting there! But doesn't seem to build according to `nix-shell -p nixpkgs-review --run "nix-review pr 187052"` which gives `error: undefined variable 'libintlperl'`

Also see some minor nitpicks inline

File: pkgs/top-level/perl-packages.nix (hunk 2 above)
Comment (qbit) [reply pkgs/top-level/perl-packages.nix]: Updated :D

File: pkgs/top-level/perl-packages.nix (hunk 4 above)
Comment (qbit) [reply pkgs/top-level/perl-packages.nix]: Fixed.

**Comment by qbit** [comment]
OK, fixed!

**Review by dasJ — APPROVED** [review]

**Comment by stigtsp** [comment]
@GrahamcOfBorg perlPackages.ZonemasterCLI

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.ZonemasterCLI

**Review by stigtsp — APPROVED** [review]
LGTM, thanks!

**Pull request merged**